### PR TITLE
BXC-1665 - Notification fixes

### PR DESCRIPTION
**Pull request by bbpennel**
https://jira.lib.unc.edu/browse/HYC-1665
* Don't send deletion approval email to managers, except when an admin performs a withdraw action directly
* Don't send "needs review" emails to depositors, only to approvers

### CODE REVIEW

File: app/services/hyrax/workflow/withdrawl_notification.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+module Hyrax
+  module Workflow
+    # Same behavior as the DeletionApprovalNotification, except that the notification is also sent
+    # to the instigating user
+    class WithdrawlNotification < DeletionApprovalNotification
```
Comment (sharonluong) [inline app/services/hyrax/workflow/withdrawl_notification.rb:6]: Withdrawal is misspelled =) can you rename it in all the locations?

File: app/overrides/services/hyrax/workflow/pending_review_notification_override.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+# https://github.com/samvera/hyrax/blame/hyrax-v3.5.0/app/services/hyrax/workflow/pending_review_notification.rb
+Hyrax::Workflow::PendingReviewNotification.class_eval do
+  private
+    # [hyc-override] Don't add the instigating user into list of users to notify
+    def users_to_notify
+      super
```
Comment (sharonluong) [inline app/overrides/services/hyrax/workflow/pending_review_notification_override.rb:7]: where is super defined?

File: spec/services/hyrax/workflow/withdrawl_notification_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  end
+  let(:workflow) do
+    Sipity::Workflow.create(name: 'test', allows_access_grant: true, active: true,
+                            permission_template_id: permission_template.id)
+  end
+  let(:entity) { Sipity::Entity.create(proxy_for_global_id: work.to_global_id.to_s, workflow_id: workflow.id) }
+  let(:comment) { double('comment', comment: 'A pleasant read') }
+
+  describe '.send_notification' do
+    it 'sends a message to depositor and user who instigated action' do
```
Comment (sharonluong) [inline spec/services/hyrax/workflow/withdrawl_notification_spec.rb:24]: the depositor doesn't seem to get a message in the following test. only the approver?

**Review by sharonluong — CHANGES_REQUESTED** [review]

File: app/services/hyrax/workflow/withdrawl_notification.rb (hunk 1 above)
Comment (bbpennel) [reply app/services/hyrax/workflow/withdrawl_notification.rb]: I have withdrawn the misspelled class name

File: app/overrides/services/hyrax/workflow/pending_review_notification_override.rb (hunk 2 above)
Comment (bbpennel) [reply app/overrides/services/hyrax/workflow/pending_review_notification_override.rb:7]: Its defined by ruby, it refers to the parent class or the parent class' implementation of the same method

Comment (sharonluong) [reply app/overrides/services/hyrax/workflow/pending_review_notification_override.rb:7]: ohh right

**Review by sharonluong — APPROVED** [review]

**Pull request merged**